src/GUI/GraphPanel.java

```diff
@@ -180,31 +175,46 @@ private void drawGrid(Graphics2D g2d, int width, int height, double step) {
         }
     }
 
-    private void drawFunctions(Graphics2D g2d, int width) {
-        GeneralPath path;
+    private void drawParametricFunction(Graphics2D g2d, ParametricExpression expression) {
+        GeneralPath path = new GeneralPath();
+        double sampleLength = (expression.upperBound - expression.lowerBound) / expression.numSamples;
+
+        Vector2D startPosition = toScreenCoordinates(expression.evaluate(expression.lowerBound));
+        path.moveTo(startPosition.x, startPosition.y);
+
+        for(int i = 1; i <= expression.numSamples; i++) {
+            double t = expression.lowerBound + i * sampleLength;
+            Vector2D position = toScreenCoordinates(expression.evaluate(t));
+            path.lineTo(position.x, position.y);
+        }
 
+        g2d.setStroke(new BasicStroke(2.0f));
+        g2d.setColor(Color.WHITE);
+        g2d.draw(path);
+    }
+
+    private void drawPolynomialFunctions(Graphics2D g2d, int width) {
```

Comment:
now that I think about it, we should just have one generic 'drawParametricFunction' that takes as input a parametric function, start and end range, steps and Color and then just draws it. The only difference between Polynomials and Parametrics so far is that polynomials get their start and end coords from the screen edges. That shouldn't happen in the 'draw' function, though

Reply:
done